src/Diagnostics.DataProviders/DataProviders/KustoDataProvider.cs

```diff
@@ -57,6 +57,23 @@ public KustoDataProvider(OperationDataCache cache, KustoDataProviderConfiguratio
 
         public async Task<DataTable> ExecuteClusterQuery(string query, string requestId = null, string operationName = null)
         {
+            var matches = Regex.Matches(query, @"cluster\((?<cluster>([^\)]+))\).database\((?<database>([^\)]+))\)\.");
+            if (matches.Any())
+            {
+                foreach (Match element in matches)
+                {
+                    string targetCluster = element.Groups["cluster"].Value.Trim(new char[] { '\'', '\"' });
+                    string targetDatabase = element.Groups["database"].Value.Trim(new char[] { '\'', '\"' });                    
+
+                    if (!string.IsNullOrWhiteSpace(targetCluster) && !string.IsNullOrWhiteSpace(targetDatabase)
+                        && targetCluster.Equals("Aks", StringComparison.OrdinalIgnoreCase) && targetDatabase.Equals("AKSprod", StringComparison.OrdinalIgnoreCase))
```

Comment:
My only suggestion here is to put the cluster and database information in our runtimehost app settings. We don't want to reveal kusto URL information here to prevent future issues.

Reply:
Good point, this however is an interim test for AKS alone, once we confirm that this change did not cause any significant errors in any of the existing detectors for this partner, we will apply this extraction logic to all partners and we won't have to store any strings anywhere and the auth on kusto cluster will prevent anything bad.

I do however get your point, would you suggest I apply it to all partner queries (given that I have manually tested a few of the queries for this partner) or run with this logic for a week and then remove the strings?

Reply:
I think that really depends on how confident you feel about applying it to all partners but if you choose to keep this logic and run it for a week then I believe we need be more slightly more secure and sensitive with partner kusto information..

Reply:
I think take your call but I would like to prevent the potential for DDoS on their cluster.

Reply:
Made the change to apply this for all partners. We will have partners test their detectors on our staging environment to confirm we do not end up breaking them.